Hyper-VMigrationJobScript.ps1

```diff
@@ -0,0 +1,166 @@
+
+        $OtherServers = @($VimTarget)
+
+        # Create a workload
+        $DtWorkloadGUID = New-DtWorkload -ServiceHost $DtSource -WorkloadTypeName $DtWorkloadType
+
+        # This workload, by default, selects all volumes for protection
+        # If desired, exclude any volumes from protection, however, be careful
+        # when excluding data as it may compromise the integrity of your installed applications
+        # Uncomment and use the following line, substituting G:\for the volume you want to exclude
+        # Repeat the line to exclude multiple volumes
+        # Set-DtLogicalItemSelection -ServiceHost $DtSource -WorkloadId $DtWorkloadGuid -LogicalPath "G:\" -Unselect
+
+        # Get the workload definition including the workload and logical items
+        $DtWorkload = Get-DtWorkload -ServiceHost $DtSource -WorkloadId $DtWorkloadGUID
+
+        # Get the default job options that will be used to create the job
+        $DtJobOptions = Get-DtRecommendedJobOptions -ServiceHost $DtTarget -Source $DtSource -JobType $DtJobType -Workload $DtWorkload -OtherServers $OtherServers
+
+        # Create the job
+        $timestamp = Get-Date -Format "yyyyMMdd.HHmmss"
+        # Generate a unique replica name
+        Write-Host "Usable tags for replica name:
+        {YYYY} = 4 digit year
+        {MM} = 2 digit month
+        {DD} = 2 digit day
+        {HH} = 2 digit hour
+        {MN} = 2 digit minutes
+        {SS} = 2 digit second
+        {MS} = milliseconds
+        {IP} = Current IP of the VM to be migrated
+        If no name is entered, a default name will be generated in the following format d42-carbonite-[source ip].[YYYYMMDD.HHMMSS]"
+        $UserReplicaName = Read-Host -Prompt 'Please enter the replica name to be created on the Hyper-V server'
+        if (!$UserReplicaName) {
+            $DtJobOptions.JobOptions.VRAOptions.ReplicaVmInfo.DisplayName = "d42-carbonite-$DtSourceName.$timestamp"
+        }  
+        else {
+            $UserReplicaName = $UserReplicaName.
+            Replace("{YYYY}" , (Get-Date -Format "yyyy")).
+            Replace("{HH}", (Get-Date -Format "HH")).
+            Replace("{MM}", (Get-Date -Format "MM")).
+            Replace("{DD}", (Get-Date -Format "dd")).
+            Replace("{MN}", (Get-Date -Format "mm")).
+            Replace("{MS}", (Get-Date -Format "ms")).
+            Replace("{SS}", (Get-Date -Format "ss")).
+            Replace("{IP}", $DtSourceName)
+        }    
+   
+        # Make the failover start right after mirroring completes (0 - auto start, 1 - manual start)
+        $DtJobOptions.JobOptions.CoreMonitorOptions.MonitorConfiguration.ProcessingOptions = 0        
+        # Change the initial HyperV startup memory in case of memory errors from Carbonite Console (in bytes)
+        $DtJobOptions.JobOptions.VRAOptions.ReplicaVMInfo.Memory = 4096000000
```

Comment:
I didn't see these 2 lines in the ESX script.  Do they need to be there?

Reply:
The memory line is applicable to HyperV only, as the job creation process errors out without a value being set.

The ShouldShutdownSource line is for debugging, I've commented it out.